src/Navigation/Plugins/recastJSPlugin.ts

```diff
@@ -500,6 +501,11 @@ export class RecastJSCrowd implements ICrowd {
      */
     private _onBeforeAnimationsObserver: Nullable<Observer<Scene>> = null;
 
+    /**
+     *  Fires each time an agent is in reach radius of his destination
+     */
+    public onReachTargetObservable = new Observable<{agentIndex: number, destination: Vector3}>();

```

Comment:
Don't forget to clear the observable when disposed (otherwise it might not be garbage-collected)

Reply:
Good catch!